rpcs3/Emu/Cell/lv2/sys_game.cpp

```diff
@@ -40,3 +49,50 @@ u64 _sys_game_get_system_sw_version()
 {
 	return stof(utils::get_firmware_version()) * 10000;
 }
+
+error_code _sys_game_watchdog_start(u64 time)
+{
+	sys_game.trace("sys_game_watchdog_start(time=%d)", time);
+
+	auto watchdog = [](u64 time)
+	{
+		while (!watchdog_stopped)
+		{
+			if (get_timestamp() - watchdog_last_clear > time * 1000 * 1000)
+			{
+				watchdog_stopped = true;
+				if (!Emu.IsStopped())
+				{
+					sys_game.warning("Watchdog timeout! Restarting the game...");
+					std::thread([](){Emu.Restart();}).detach();
```

Comment:
if this is already in a thread, why does it need another?

Reply:
Aha, right.